### PR TITLE
feat(swift): add disjunctive faceting

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-2852]

### Changes included:

- disjunctive faceting

## 🧪 Test

Unit tests imported from working v8

[DI-2852]: https://algolia.atlassian.net/browse/DI-2852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/algoliasearch-client-swift/Sources/Search/Extra/SearchQueryExtension.swift [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  }
+  
+  public init(from searchParamsObject: SearchSearchParamsObject, options: SearchForFacetsOptions, params: String? = nil) {
+    self = .searchForFacets(SearchForFacets(from: searchParamsObject, options: options, params: params))
+  }
+}
+
+extension SearchForHits {
+  public init(from searchParamsObject: SearchSearchParamsObject, indexName: String, params: String? = nil) {
+    self.params = params
```
Comment (shortcuts) [inline clients/algoliasearch-client-swift/Sources/Search/Extra/SearchQueryExtension.swift:20]: is this generated? D:

File: clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift [hunk 2]
```diff
@@ -637,4 +637,39 @@ public extension SearchClient {
...
     }
+  
+  /**
+   Method used for perform search with disjunctive facets.
+   
+   - Parameter indexName: The name of the index in which the search queries should be performed
+   - Parameter searchParamsObject: The search query params.
+   - Parameter refinements: Refinements to apply to the search in form of dictionary with
+     facet attribute as a key and a list of facet values for the designated attribute
+   - Parameter disjunctiveFacets: Set of facets attributes applied disjunctively (with OR operator)
```
Comment (shortcuts) [inline clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift:648]: Might be worth to state that anything not in this list, but in `refinements` will be conjunctively computed

File: clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import Foundation
+
+public struct SearchDisjunctiveFacetingResponse<T: Codable> {
+  let response: SearchResponse<T>;
+  let disjunctiveFacets: [String: [String: Int]];
+}
+
+/// Helper making multiple queries for disjunctive faceting
+/// and merging the multiple search responses into a single one with
+/// combined facets information
```
Comment (shortcuts) [inline clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift:17]: I don't know swift so ignore if it's incorrect, but if this helper is exposed in the final build, can we add a disclaimer that we expect this to be used by `searchDisjunctiveFaceting`?

File: clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+public enum DisjunctiveFacetingError: Error, LocalizedError {
+
+  case emptyResponses
+
+  var localizedDescription: String {
+    switch self {
+    case .emptyResponses:
+      return "Unexpected empty search responses list. At least one search responses might be present."
```
Comment (shortcuts) [inline clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift:150]: wouldn't this be caught by the transporter? I'm not sure it's possible to have 0 responses

**Review by shortcuts — COMMENTED** [review]
mostly reviewed the general implem as I'm not a swifty master like you

File: clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift (hunk 3 above)
Comment (Fluf22) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift:17]: Made it internal so they don't use it directly
Only the API method is publicly accessible now

File: clients/algoliasearch-client-swift/Sources/Search/Extra/SearchQueryExtension.swift (hunk 1 above)
Comment (millotp) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/SearchQueryExtension.swift:20]: it's _AI Generated_

Comment (Fluf22) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/SearchQueryExtension.swift:20]: Hahaha

File: clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift (hunk 2 above)
Comment (Fluf22) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift:649]: Yes, being more explicit won't hurt

File: clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift (hunk 3 above)
Comment (shortcuts) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/DisjunctiveFaceting.swift:17]: nicee

**Review by shortcuts — APPROVED** [review]
nothing to add, you clou'd it

**Pull request merged**